hs_core/views/__init__.py

```diff
@@ -252,8 +254,9 @@ def delete_resource(request, shortkey, *args, **kwargs):
 
     return HttpResponseRedirect('/my-resources/')
 
+
 def create_new_version_resource(request, shortkey, *args, **kwargs):
-    res, authorized, user = authorize(request, shortkey, edit=True, full=True, superuser=True, raises_exception=False)
+    res, authorized, user = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.CREATE_RESOURCE_VERSION)
     if res.raccess.published and not authorized:
         if user.uaccess.owns_resource(res):
             authorized = True
```

Comment:
@pkdash Using the new authorize() implementation, lines 260-262 above are not needed any more which were added due to the old authorize() implementation does not account for published status which keeps ownership but does not allow owners to edit or delete the published resource. With the new authorize() implementation, they are not needed. So please remove these three lines above.